multihost_test/bz_automation/script/myxauth.c

```diff
@@ -0,0 +1,35 @@
+/*
+ * Print real and effective user ID
+ *
+ *
+ *
+ */
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h> 
+#include <sys/types.h>
+
+int main(int argc, char**argv) 
+{
+    FILE *file;
+    char *fname="/tmp/xauthlog";
+    int i;
+
+    file = fopen("/tmp/xauthlog", "a+");
+    if(file == NULL) {
+        printf("Failed creating file /tmp/xauthlog");
+        return -1;
+    }
+
+    printf("NECHsaPACI");
+
+    fprintf(file, "Args: ");
+    for(i = 0; i < argc-1; i++) fprintf(file, "%s ", argv[i]);
+    fprintf(file, "\n");
+
+    fprintf(file, "%d/%d\n", getuid(), geteuid()); 
+    fclose(file);
+    system("chmod 0777 /tmp/xauthlog");
+    //chmod("/tmp/xauthlog", 0777); 
```

Comment:
Why don't you call directly `chmod()`?

Reply:
I was getting error:

[root@client1 ~]# gcc myauth.c 
myauth.c: In function ‘main’:
myauth.c:33:5: warning: implicit declaration of function ‘chmod’; did you mean ‘chroot’? [-Wimplicit-function-declaration]
     chmod("/tmp/xauthlog", 0777);
     ^~~~~
     chroot
[root@client1 ~]#

Reply:
You need to `#include <sys/stat.h>`

Reply:
Done